### PR TITLE
Basic rkt support in calicoctl

### PR DESCRIPTION
**Pull request by caseydavenport**

### CODE REVIEW

File: calicoctl/calico_ctl/diags.py [hunk 1]
```diff
@@ -135,9 +145,11 @@ def save_diags(log_dir):
             f.writelines("Error running ipset: %s\n" % e)
 
     # Ask Felix to dump stats to its log file - ignore errors as the
-    # calico-node might not be running
-    subprocess.call(["docker", "exec", "calico-node",
-                     "pkill", "-SIGUSR1", "felix"])
+    # calico-node might not be running.  If using rkt runtime, skip this step, 
+    # since we can't yet run commands inside a rkt fly pod.
```
Comment (tomdee) [inline calicoctl/calico_ctl/diags.py]: We could just `pkill -SIGUSR1 felix` from the host? And that would work on docker too

File: calicoctl/calico_ctl/diags.py [hunk 2]
```diff
@@ -135,9 +145,11 @@ def save_diags(log_dir):
...
+    if runtime == "docker":
+        subprocess.call(["docker", "exec", "calico-node",
+                         "pkill", "-SIGUSR1", "felix"])
 
     if os.path.isdir(log_dir):
         print("Copying Calico logs")
```
Comment (tomdee) [inline calicoctl/calico_ctl/diags.py]: When running under rkt, the calico-node logs will be in the journal and not on the filesystem. Should we call out to the journal instead?

File: calicoctl/calico_ctl/status.py [hunk 3]
```diff
@@ -99,6 +100,57 @@ def status(arguments):
...
+                                  docker_client.containers())
+        if len(calico_node_info) == 0:
+            print "calico-node container not running"
+            sys.exit(1)
+        else:
+            print "calico-node container is running. Status: %s" % \
+                  calico_node_info[0]["Status"]
+
+            libraries_cmd = docker_client.exec_create("calico-node",
+                                                      ["sh", "-c",
```
Comment (tomdee) [inline calicoctl/calico_ctl/status.py]: Not sure why we have this `sh -c`

File: calicoctl/calico_ctl/status.py [hunk 4]
```diff
@@ -99,6 +100,57 @@ def status(arguments):
...
+                                                       "cat libraries.txt"])
+            libraries_out = docker_client.exec_start(libraries_cmd)
+            result = re.search(r"^calico\s*\((.*)\)\s*$", libraries_out,
+                               re.MULTILINE)
+
+            if result is not None:
+                print "Running felix version %s" % result.group(1)
+    except ConnectionError:
+        print "Docker is not running"
+        # TODO: Perform status checks in platform-independent way.
```
Comment (tomdee) [inline calicoctl/calico_ctl/status.py]: Could we remove this TODO now that we have a --runtime option?

File: calicoctl/calico_ctl/status.py [hunk 5]
```diff
@@ -99,6 +100,57 @@ def status(arguments):
...
+        sys.exit(1)
+
+
+def check_container_status_rkt():
+    """
+    Checks and prints the calico/node container status when running in rkt. 
+    """
+    list_cmd = ["sudo", "rkt", "list"]
+    p = Popen(list_cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE)
```
Comment (tomdee) [inline calicoctl/calico_ctl/status.py]: What's wrong with just doing check_output?

**Comment by tomdee** [comment]
Finished looking at this - a few comments/questions.

File: calicoctl/calico_ctl/status.py (hunk 5 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/status.py]: It doesn't capture stderr should something go wrong.  This let's us handle the error better.

File: calicoctl/calico_ctl/status.py (hunk 4 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/status.py]: Yes.

File: calicoctl/calico_ctl/status.py (hunk 3 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/status.py]: Me either.

File: calicoctl/calico_ctl/diags.py (hunk 2 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/diags.py]: d'oh, forgot that bit..

File: calicoctl/calico_ctl/diags.py (hunk 1 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/diags.py]: Done.

File: calicoctl/calico_ctl/diags.py (hunk 2 above)
Comment (caseydavenport) [inline calicoctl/calico_ctl/diags.py]: Done.

**Comment by caseydavenport** [comment]
Should be ready for another look @tomdee

**Comment by tomdee** [comment]
LGTM after you fix UT failures

**Pull request merged**